src/benchmarks/beluga_vs_nav2_multi_dataset/reports/nominal_report/sections/main_corpus.inc

```diff
@@ -0,0 +1,167 @@
+
+    bagfile_location_data = data.groupby([
+        'case_name',
+        'bagfile',
+    ])['basedir'].agg(['first']).rename(columns={'first': 'basedir'})
+
+    from ament_index_python import get_package_share_directory
+
+    this_pkg_dir = get_package_share_directory('beluga_vs_nav2_multi_dataset')
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_table_header.txt', 'r') as f:
+        TABLE_HEADER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_table_row.txt', 'r') as f:
+        ROW_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_dataset_chapter.txt', 'r') as f:
+        DATASET_CHAPTER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_bagfile_section.txt', 'r') as f:
+        BAGFILE_SECTION_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_resource_table_header.txt', 'r') as f:
+        RESOURCE_TABLE_HEADER_TEMPLATE = f.read()
+
+    with open(f'{this_pkg_dir}/reports/nominal_report/templates/template_resource_table_row.txt', 'r') as f:
+        RESOURCE_TABLE_ROW_TEMPLATE = f.read()
+
+    output = []
+
+    def equalize_digits(value, digits=3):
+        return f'{value:.{digits}f}' if value is not str else value
+
+    desired_section_order = [
+        'Magazino Datasets',
+        'Openloris Office',
+        'TorWIC Mapping',
+        'TorWIC SLAM',
+        'Willow Garage',
+        'Diff Drive Sim 24hs',
+        'Omni Drive Sim 24hs',
+    ]
+    datasets_in_data = ape.index.get_level_values(0).unique()
+    missing_datasets = [x for x in datasets_in_data if x not in desired_section_order]
+    desired_section_order = desired_section_order + missing_datasets
+
+    for case_name in desired_section_order:
+        output.append(DATASET_CHAPTER_TEMPLATE.format(dataset_name=case_name))
+        case_name_str = case_name.replace(' ', '_').lower()
+        output.append(f'.. include:: sections/datasets/{case_name_str}_description.inc\n\n')
+        data_for_case = ape.loc[case_name]
+        for bagfile in data_for_case.index.get_level_values(0).unique():
+            metadata = ""
+            mapfile = ""
+            try:
+                bagpath = bagfile_location_data.loc[case_name].loc[bagfile].loc['basedir']
+                metadata = get_bagfile_info(os.path.join('..', '..', bagpath, 'bagfiles', bagfile))
+                metadata = [x for x in metadata.split('\n') if "Topic:" not in x]
+                metadata = "    " + "\n    ".join(metadata)
+                mapfile = os.path.join('..', '..', bagpath, 'artifacts', bagfile, 'map.pgm')
+            except:
+                pass
+            output.append(BAGFILE_SECTION_TEMPLATE.format(bagfile_name=bagfile, metadata=metadata))
+            data_for_bagfile = data_for_case.loc[bagfile]
+            try:
```

Comment:
@glpuga why so many `try..except` blocks?

Reply:
Becase lambkin frequently fails a test case, and yet moves on leaving a hole in the data. Without this, even a single instance of that would prevent me from creating the output pdf. I use this to try to power-through and just display "???" in the tables that have missing data.

Reply:
Hmm, doesn't test case failure stop the entire benchmark? :thinking: 

About holes in data. Maybe backfilling the dataset with NaNs or "" could be a useful idiom to make report generation robust to flakey benchmarks. We don't really support that use case well.

Reply:
> Hmm, doesn't test case failure stop the entire benchmark? 🤔

Most of the time it does, but for some reason not always.

Reply:
Notice that LAMBKIN **not stopping** is the preferred behavior.

LAMBKIN should be fault-tolerant, and continue to do as much work as possible, while tagging bad test cases, and allowing to rerun those that failed.

It's no good if a four-day run fails on Saturday at 6am, and the user figures on Monday.

Reply:
Hmm, that's a very good point. RobotFramework has some provisions for [_continue on failure_ behavior](https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#continuing-on-failure), but there doesn't seem to be a global switch we can flip. Sounds like a feature request to me.